### PR TITLE
contrib: Setup all dependencies in deps/ directory.

### PR DESCRIPTION
**Pull request by mpreiner**
Every dependency set up via the corresponding get-* script now downloads
the sources to deps/ and is installed into deps/install. This
makes the cmake/Find* modules simpler and allows us to cache deps/install
for our CI builds.

### CODE REVIEW

File: contrib/get-cryptominisat [hunk 1]
```diff
@@ -2,28 +2,25 @@
...
-
 cmake -DENABLE_PYTHON_INTERFACE=OFF \
       -DSTATICCOMPILE=ON \
       -DNOM4RI=ON \
+      -DONLY_SIMPLE=ON \
+      -DCMAKE_INSTALL_PREFIX="$INSTALL_DIR" \
       ..
 
-make libcryptominisat5 -j$(nproc)
+make libcryptominisat5 -j$(nproc) install
```
Comment (4tXJ7f) [inline contrib/get-cryptominisat]: Won't the install build everything? I.e. wouldn't it be the same as just writing `make install -j$(nproc)`? Can we still install CMS without Boost?

File: contrib/get-script-header.sh [hunk 2]
```diff
@@ -2,7 +2,15 @@
 #
 set -e -o pipefail
 
-cd "$(dirname "$0")/.."
+[ ! -e contrib ] && echo "$0 not called from base directory" && exit 1
```
Comment (4tXJ7f) [inline contrib/get-script-header.sh]:
```suggestion
[ ! -d contrib ] && echo "$0 not called from base directory" && exit 1
```

File: contrib/get-script-header.sh [hunk 3]
```diff
@@ -36,3 +44,52 @@ if [ -z "$PYTHON" ]; then
...
+  cp "$1" "$INSTALL_LIB_DIR"
+}
+
+function install_includes
+{
+  include="$1"
+  subdir="$2"
+  echo "Copying $1 to $INSTALL_INCLUDE_DIR/$subdir"
+  [ ! -d "$INSTALL_INCLUDE_DIR" ] && mkdir -p "$INSTALL_INCLUDE_DIR"
+  [ -n "$subdir" ] && mkdir -p "$INSTALL_INCLUDE_DIR/$subdir"
```
Comment (4tXJ7f) [inline contrib/get-script-header.sh]:
```suggestion
  [ -n "$subdir" ] && [ ! -d "$INSTALL_INCLUDE_DIR/$subdir" ] && mkdir -p "$INSTALL_INCLUDE_DIR/$subdir"
```

For consistency?

File: contrib/get-cryptominisat (hunk 1 above)
Comment (mpreiner) [reply contrib/get-cryptominisat]: With `-DONLY_SIMPLE=ON` it builds CMS without boost. (https://github.com/msoos/cryptominisat/blob/master/CMakeLists.txt#L429)

**Pull request merged**